nevergrad/optimization/optimizerlib.py

```diff
@@ -1815,6 +1816,139 @@ def __init__(
+                obj_fun=None,  # Assuming that this is not used :-)
+                DoE_size=init_budget if init_budget is not None else 5,
+                max_FEs=budget,
+                verbose=True,
+                n_point=1,  # We start with a sequential procedure, maybe we'll extend in a second moment
+                n_components=n_components,
+                acquisition_optimization={"optimizer": "BFGS"},
+            )
+        else:
+            self._alg = BayesOptimBO(
+                search_space=space,
+                obj_fun=None,  # Assuming that this is not used :-)
+                model=model,
+                DoE_size=init_budget if init_budget is not None else 5,
+                max_FEs=budget,
+                verbose=True,
+            )
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        if not self._buffer:
+            self._newX = []
+            self._losses = []
+            candidate = self._alg.ask()
+            if not isinstance(candidate, list):
+                candidate = candidate.tolist()
+            self._buffer = candidate
+        x_probe = self._buffer.pop()
+        data = np.tan(np.array(x_probe, copy=False))
+        candidate = self.parametrization.spawn_child().set_standardized_data(data)
+        candidate._meta["x_probe"] = x_probe
+        self._newX.append(candidate._meta["x_probe"])
+        return candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        self._losses.append(loss)
+        if not self._buffer:
+            if "x_probe" in candidate._meta:
+                self._alg.tell(self._newX, self._losses)
+            else:
+                data = candidate.get_standardized_data(reference=self.parametrization)
+                # Tell not asked:
+                self._alg.tell(np.arctan(data), loss)
+
+    def _internal_tell_not_asked(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        raise errors.TellNotAskedNotSupportedError
+
+
+class ParametrizedBayesOptimBO(base.ConfiguredOptimizer):
+    """
+    Algorithms from bayes-optim package.
+
+    We use:
+    - BO
+    - PCA-BO: Principle Component Analysis (PCA) Bayesian Optimization for dimensionality reduction in BO
+
+    References
+
+    [RaponiWB+20]
+        Raponi, Elena, Hao Wang, Mariusz Bujny, Simonetta Boria, and Carola Doerr.
+        "High dimensional bayesian optimization assisted by principal component analysis."
+        In International Conference on Parallel Problem Solving from Nature, pp. 169-183.
+        Springer, Cham, 2020.
+
+    """
+
+    no_parallelization = True
+    recast = True
+
+    # pylint: disable=unused-argument
+    def __init__(
+        self,
+        *,
+        # num_workers: int = 1,
+        init_budget: tp.Optional[int] = None,
+        pca: tp.Optional[bool] = False,
+        n_components: tp.Optional[float] = 0.95,
+        prop_doe_factor: tp.Optional[float] = None,
```

Comment:
can you explain describe the parameters in the docstrings?
See other examples, in this file there should be many (look for "Parameters")

Reply:
Done